server/public/model/config.go

```diff
@@ -4328,49 +4328,124 @@ func (o *Config) GetSanitizeOptions() map[string]bool {
 	return options
 }
 
-func (o *Config) Sanitize() {
-	if o.LdapSettings.BindPassword != nil && *o.LdapSettings.BindPassword != "" {
-		*o.LdapSettings.BindPassword = FakeSetting
+type SanitizeConfigRules struct {
+	Contains   []string
+	StartsWith []string
+	EndsWith   []string
+	Whitelist  []string
```

Comment:
I'd probably rename to `AllowList`.

Also, `Contains` being a super-set, so we could use that as well as a general case of exact matches.